### PR TITLE
WIP: Docs/global/document global variables

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:

**Special notes for your reviewer**:
I migrated our main docs to the demo. Our docs is seriously behind and needs to be refactored. Please checkout how stuff looks, both in design and how it's implemented and let me know what you think.

The branch is built on top of the slots PR. So ignore the diffs related to that one for now.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by mpourismaiel** [comment]
I thought I had moved it all except for the doc on the contact fragment and that was due to my laptop being low on battery :D Is there a part I missed?

I wanted to propose removing the previous `docs/` directory in favor of these pages. What do you think?

I'm going to rewrite the places that need changing and write new docs on slots, more on global variables and how to write fragment controller files and provide more examples since we have more control inside our demo pages. It may take some time from tomorrow as well.

File: exampleSite/content/docs/image-fallthrough/content.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+- If the content file controlling the fragment is located in it's own directory (`content/[page]/[fragment]/[filename].md`), fragment will look for a file with the name specified in that content in that directory.
+- If the specified file is not found in that directory or the controlling content is in page directory, fragment will look in that page directory as well.
+- If the file is not found in the page directory the fragment will look in images directory for that file.
+
+So the fragment will look in the following order `fragment > page > images (global)`. If you need to use an image in several pages you can put it in the `static/images/` directory and the image would be available globally. But if an image may differ between two pages or even two fragments of same type, it's possible to have it using this mechanism.
+
+Syna supports custom favicons in config.toml allowing for ICO, PNG or SVG image formats. In order to use one of the custom favicon formats, you can specify the image file name in config.toml and save the image file in the '/static' directory.
```
Comment (stp-ip) [inline exampleSite/content/docs/image-fallthrough/content.md]:
``` suggestion
Syna supports custom favicons in config.toml allowing for ICO, PNG or SVG image formats. In order to use one of the custom favicon formats, you can specify the image file name in config.toml and save the image file in the 'static/' directory.
```

File: exampleSite/content/docs/supported-colors/content.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| primary   | #00838F |
+| secondary | #868e96 |
+| success   | #008f54 |
+| info      | #00838F |
+| warning   | #fdf314 |
+| danger    | #dc1200 |
+| light     | #f8f9fa |
+| dark      | #343a40 |
+
+Classes can be applied to style text, buttons and fragment backgrounds and links.
```
Comment (stp-ip) [inline exampleSite/content/docs/supported-colors/content.md]:
``` suggestion
Classes are mainly used for backgrounds and button colors.  
Links, inline code and other elements will also be changed.
```

File: exampleSite/content/docs/supported-colors/content.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+These colors can also be overwritten for more details see our [style documentation](#styles).
```
Comment (stp-ip) [inline exampleSite/content/docs/supported-colors/content.md]:
``` suggestion
These colors can also be overwritten for more details see our [style documentation](/docs/development/#styles).
```

File: exampleSite/content/docs/fragments/content.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+Global fragments are located in a special content directory `content/_global/`.
+All fragments within this directory are rendered on all pages by default.
+*To not render the whole page as a subpath of your website the index.md file defines the whole directory as a `headless` bundle*
+To overwrite a global fragment create a per page fragment with the same filename.
+This would overwrite the global one.
+
+#### How to control fragments
+
+Each fragment is controlled by a content file, which is usually located next to the page's index.md file.  
+**content/my-page/index.md** *defines the page and a few attributes such as page title*
```
[the remark this answers is not in the document]
Comment (mpourismaiel) [reply exampleSite/content/docs/fragments/content.md]: The docs is extracted from the old `docs/readme.md`. It was bold in there. Should I change it?

**Comment by stp-ip** [comment]
Why do we use blockquote for the header? That feels strange.
Besides that I agree to remap this.

**Comment by stp-ip** [comment]
Also the ToC on top of the page still feels imperfect.
* think the bullet points might be too much
* padding is too much

Thoughts?

**Comment by stp-ip** [comment]
That looks good. Feels cleaner without the bullet points. That would also improve the toc at the top of the content fragment.

Still trying to figure out, why we do blockquote instead of just using a heading. ^^

**Comment by mpourismaiel** [comment]
Can't remember why we did it in the old docs.

**Comment by stp-ip** [comment]
Ok so let's not do this in the new ones, but still keep the blockquote "fix".

**Comment by mpourismaiel** [comment]
Removed the bullets for now and changed the left side padding. Will add an issue for making the list fragment and toc fragment collapsable.

**Pull request merged**